starter_code/src/movies.js

```diff
@@ -1,19 +1,108 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(moviesArr) {
+    let avgRate =
+        moviesArr.reduce((accumulator, movie) => {
+            if (!movie.rate) return accumulator;
+            return accumulator + parseFloat(movie.rate);
+        }, 0) / moviesArr.length;
+    avgRate = parseFloat(avgRate.toFixed(2));
+    return avgRate;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
-
-
+function dramaMoviesRate(moviesArr) {
+    const dramaSelection = moviesArr.filter(movie =>
+        movie.genre.includes("Drama")
+    );
+    if (!dramaSelection.length) return 0;
+    return ratesAverage(dramaSelection);
+}
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+function orderByDuration(moviesArr) {
+    const moviesCopy = moviesArr.slice();
+    if (moviesArr.length === 1) return moviesArr;
 
+    //
+    for (movie of moviesCopy) {
+        if (typeof movie.duration === "string") {
```

Comment:
this still mutates the object in the original array, use the `...` operator to spread all the properties of the original object